### DISCUSSION TITLE
Provide a way to know which interpreter `py` would use

### DISCUSSION THREAD

**Discussion by SamEdwardes** [brettcannon/python-launcher#226] [Ideas]
Is there any way to check what interpreter `py` will execute?

For example, I imagine a flow like this:

```bash
$ cd ~/my-project
$ py --which
/opt/python/3.10.0/bin/python
$ py -m venv .venv
$ py --which
~/my-project/.venv/bin/python
```

**Comment by brettcannon** [comment]
There's nothing currently for this. The closest that's been thought about is https://github.com/brettcannon/python-launcher/discussions/74, but not a specific command. I've gone ahead and turned this into an idea/feature request.

Can I ask why you specifically want this instead of doing `py -c "import sys; print(sys.executable)`?

**Reply by SamEdwardes** [reply]
I never thought of using `py -c "import sys; print(sys.executable)"` - thanks for sharing this idea! My idea for `py --which` is inspired by the `pyenv which python` command from pyenv which I find helpful to figure out where the location of the actual interpreter is. I think having a `py --which` or `py which` command is just nicer than typing `py -c "import sys; print(sys.executable)"`.

**Comment by brettcannon** [comment]
This could be solved via https://github.com/brettcannon/python-launcher/discussions/222 or https://github.com/brettcannon/python-launcher/discussions/221 .

**Comment by brettcannon** [comment]
One issue with this idea is #168 and how not all things can be simplified down to a single path, e.g. conda environments would need to be either run from an activated terminal or used via `conda run`. In that scenario this may need to error out to say that there isn't a way to run the interpreter as just a path alone.

**Comment by brettcannon** [comment]
This will be handled via #432 .
